### DISCUSSION TITLE
Forced anonymous volume for `/data` - any way around that?

### DISCUSSION THREAD

**Discussion by subrezon** [itzg/docker-minecraft-server#2876] [Q&A] [answered]
Hi everyone, I have a couple of use cases where I would like to only mount volumes for `/data/world` and sometimes `/data/backups`, but redownload/recreate the rest of `/data` on each start. However, the image force creates an anonymous volume on container creation, meaning that instead of stopping/starting the containers I have to remove/recreate them.

I understand why it's there, I'm likely the only one complaining and it probably helps Docker beginners a lot. Is there a workaround I can use for this?

**Comment by itzg** [comment, marked as answer]
That would be a general docker question since I don't know if there's a way to un-VOLUME the declaration. 

Forking the repo and building your own image would be the easiest. I would have suggested changing the working directory but unfortunately I hard coded /data in half the places. An enhancement to search and replace that with a  variable might help with that. 

A wrapper script around the whole docker setup would easily automate away the volume cleanup too.

**Reply by subrezon** [reply]
Okay, thanks a lot! I had hoped there would be an easier way, I'll just fork the repo and remove the VOLUME declaration from the Dockerfile.
